src/components/AddComment.jsx

```diff
@@ -0,0 +1,66 @@
+import { useContext, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { UserContext } from "../contexts/UserContext";
+import { postComment } from "../utils/api";
+
+const AddComment = ({ review_id, setComments, setSelectedReview, comments }) => {
+    
+    const [newComment, setNewComment] = useState('');
+    const [loading, setLoading] = useState(false);
+    const [submitted, setSubmitted] = useState(false);
+
+    const navigate = useNavigate();
+    
+    const userValue = useContext(UserContext);
+    const loggedInUser = userValue.loggedInUser.username;
+
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        if(newComment.length === 0) {
+            alert('Comment box is empty!');
+            setLoading(false);
+        }         
+        else if(loggedInUser && newComment.length > 0) {
+            postComment(newComment, review_id, loggedInUser, setLoading)
```

Comment:
We would need to add something for the user experience if the post request to the api fails, think about what you would expect to see if a comment post failed?

Reply:
added catch block